components/count-badge/count-badge-mixin.js

```diff
@@ -0,0 +1,189 @@
+
+	static get properties() {
+		return {
+			/**
+			 * Optionally choose to announce changes to the badge with an aria-live region. If the number property is changed, the text will be read by screenreaders. Defaults to false.
+			 * @type {boolean}
+			 */
+			announceChanges: {
+				type: Boolean,
+				attribute: 'announce-changes'
+			},
+			/**
+			 * Optionally add a tooltip on the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			hasTooltip: {
+				type: Boolean,
+				attribute: 'has-tooltip'
+			},
+			/**
+			 * Optionally choose to not render the count badge when the number is zero. Defaults to false.
+			 * @type {boolean}
+			 */
+			hideZero: {
+				type: Boolean,
+				attribute: 'hide-zero'
+			},
+			/**
+			 * Optionally specify a digit limit, after which numbers are truncated. Defaults to two for "notification" type and no limit for "count" type.
+			 * @type {number}
+			 */
+			maxDigits: {
+				type: Number,
+				attribute: 'max-digits'
+			},
+			/**
+			 * The number to be displayed on the badge. Must be a positive integer.
+			 * @type {number}
+			 */
+			number: {
+				type: Number,
+				attribute: 'number'
+			},
+			/**
+			 * The size of the badge. Defaults to "small".
+			 * @type {'small'|'large'}
+			 */
+			size: {
+				type: String,
+				reflect: true,
+				attribute: 'size'
+			},
+			/**
+			 * Optionally choose to add a tab stop to the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			tabStop: {
+				type: Boolean,
+				attribute: 'tab-stop'
+			},
+			/**
+			 * * Descriptive text for the badge which will act as an accessible label and tooltip text when tooltips are enabled.
+			 * @type {string}
+			 */
+			text: {
+				type: String
+			},
+			/**
+			 * The type of the badge. Defaults to "count".
+			 * @type {'count'|'notification'}
+			 */
+			type: {
+				type: String,
+				reflect: true,
+				attribute: 'type'
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this.announceChanges = false;
+		this.hasTooltip = false;
+		this.hideZero = false;
+		this.size = 'small';
+		this.tabStop = false;
+		this.text = '';
+		this.type = 'count';
+		this._textId = getUniqueId();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.maxDigits && this.type === 'notification') {
+			// default to two digits for notification type
+			this.maxDigits = 2;
+		}
+	}
+
+	getOffscreenId() {
+		return this.hasTooltip ? undefined : this._textId;
+	}
+
+	renderCount(numberStyles) {
+		let numberString = `${this.number}`;
+		const hideNumber = this.hideZero && this.number === 0;
+		if (hideNumber) {
+			numberString = '';
+		}
+		numberStyles = {
+			...numberStyles,
+			visibility: hideNumber ? 'hidden' : 'visible'
+		};
+		if (this.maxDigits && this.number.toString().length > this.maxDigits) {
+			numberString = `${'9'.repeat(this.maxDigits)}`;
+			numberString = formatNumber(parseNumber(numberString));
```

Comment:
It's confusing/weird, but `parseNumber` is expecting the number in the user's locale, whereas you're creating a number by just repeating `9` a bunch of times, so you know that it's just a "normal" number -- the built-in native `parseInt()` API is meant for that.
```suggestion
			numberString = formatNumber(parseInt(numberString));
```

Reply:
Ah I see, will make that change. Too many different ways to write numbers... 🤦